lib/training_data.py

```diff
@@ -27,3 +27,61 @@ def get_training_data(images, batch_size):
         target_images[i] = target_img
 
     return warped_images, target_images
+
+def random_transform(image, rotation_range, zoom_range, shift_range, random_flip):
+    h, w = image.shape[0:2]
+    rotation = numpy.random.uniform(-rotation_range, rotation_range)
+    scale = numpy.random.uniform(1 - zoom_range, 1 + zoom_range)
+    tx = numpy.random.uniform(-shift_range, shift_range) * w
+    ty = numpy.random.uniform(-shift_range, shift_range) * h
+    mat = cv2.getRotationMatrix2D((w // 2, h // 2), rotation, scale)
+    mat[:, 2] += (tx, ty)
+    result = cv2.warpAffine(
+        image, mat, (w, h), borderMode=cv2.BORDER_REPLICATE)
+    if numpy.random.random() < random_flip:
+        result = result[:, ::-1]
+    return result
+
+# get pair of random warped images from aligened face image
+
+def random_warp(image):
+    assert image.shape == (256, 256, 3)

```

Comment:
Probably want to make this assertion more explicit, so the user knows what's wrong with his image. It seems to fail silently due to us catching all exceptions at the beginning of `faceswap.py` and any attempt to parse it comes up without error message.